plugins/georeplication/rest.go

```diff
@@ -0,0 +1,193 @@
+package georeplication
+
+import (
+	errs "errors"
+	"net/http"
+
+	"github.com/gluster/glusterd2/errors"
+	"github.com/gluster/glusterd2/gdctx"
+	georepapi "github.com/gluster/glusterd2/plugins/georeplication/api"
+	restutils "github.com/gluster/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/transaction"
+	"github.com/gluster/glusterd2/utils"
+	"github.com/gluster/glusterd2/volume"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
+)
+
+func validateMasterAndSlaveIDFormat(w http.ResponseWriter, masteridRaw string, slaveidRaw string) (uuid.UUID, uuid.UUID, error) {
+	// Validate UUID format of Master and Slave Volume ID
+	masterid := uuid.Parse(masteridRaw)
+	if masterid == nil {
+		restutils.SendHTTPError(w, http.StatusBadRequest, "Invalid Master Volume ID")
+		return nil, nil, errs.New("Invalid Master Volume ID")
+	}
+
+	// Validate UUID format of Slave Volume ID
+	slaveid := uuid.Parse(slaveidRaw)
+	if slaveid == nil {
+		restutils.SendHTTPError(w, http.StatusBadRequest, "Invalid Slave Volume ID")
+		return nil, nil, errs.New("Invalid Slave Volume ID")
+	}
+
+	return masterid, slaveid, nil
+}
+
+func georepCreateUpdateHandler(w http.ResponseWriter, r *http.Request) {
+	// Collect inputs from URL
+	p := mux.Vars(r)
+	masteridRaw := p["mastervolid"]
+	slaveidRaw := p["slavevolid"]
+
+	reqID, logger := restutils.GetReqIDandLogger(r)
+
+	// Same handler for both Create and Update, if HTTP method is POST then
+	// it is update request
+	updateRequest := false
+	if r.Method == http.MethodPost {
+		updateRequest = true
+	}
+
+	// Validate UUID format of Master and Slave Volume ID
+	masterid, slaveid, err := validateMasterAndSlaveIDFormat(w, masteridRaw, slaveidRaw)
```

Comment:
To be consistent with rest of the code, you can return SendHTTPError() from the handler itself.

```go
 if uuid.Parse(masteridRaw) == nil || uuid.Parse(slaveidRaw) == nil {
    restutils.SendHTTPError(w, http.StatusBadRequest, "Invalid Master/Slave Volume ID")
    return
}
```
Or you can put all of the below validations (MasterVol, SlaveHosts, SlaveVol) in a single validation function. This will also make the handler function smaller and more readable.

Reply:
Initially I had that in handler itself, but observed that I have to repeat the stuff almost for all Georep API calls. (MasterVol, SlaveHosts and SlaveVol) validation is required only for Create/Update. For all the other apis Start/Stop/Delete/Status/Config we don't need to validate these fields except Master ID and Slave ID